oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/DocumentPropertyStateTest.java

```diff
@@ -113,287 +80,4 @@ public void multiValuedBinarySize() throws Exception {
-    }
-
-    @Test
-    public void stringAboveThresholdSize() throws Exception {
-        NodeBuilder builder = ns.getRoot().builder();
-        builder.child(TEST_NODE).setProperty("p", STRING_HUGEVALUE, Type.STRING);
-        TestUtils.merge(ns, builder);
-
-        PropertyState p = ns.getRoot().getChildNode(TEST_NODE).getProperty("p");
-        assertEquals(Type.STRING, Objects.requireNonNull(p).getType());
-        assertEquals(1, p.count());
-
-        reads.clear();
-        assertEquals(10050, p.size(0));
-        // must not read the string via streams
-        assertEquals(0, reads.size());
-    }
-
-    @Test
-    public void compressValueThrowsException() throws IOException, NoSuchFieldException, IllegalAccessException {
-        DocumentNodeStore mockDocumentStore = mock(DocumentNodeStore.class);
-        Compression mockCompression = mock(Compression.class);
-        when(mockCompression.getOutputStream(any(OutputStream.class))).thenThrow(new IOException("Compression failed"));
-
-        DocumentPropertyState.setCompressionThreshold(DEFAULT_COMPRESSION_THRESHOLD);
-        DocumentPropertyState documentPropertyState = new DocumentPropertyState(mockDocumentStore, "p", "\"" + STRING_HUGEVALUE + "\"", mockCompression);
-
-        assertEquals(documentPropertyState.getValue(Type.STRING), STRING_HUGEVALUE);
-
-        verify(mockCompression, times(1)).getOutputStream(any(OutputStream.class));
-    }
-
-    @Test
-    public void uncompressValueThrowsException() throws IOException, NoSuchFieldException, IllegalAccessException {
-
-        DocumentNodeStore mockDocumentStore = mock(DocumentNodeStore.class);
-        Compression mockCompression = mock(Compression.class);
-        OutputStream mockOutputStream= mock(OutputStream.class);
-        when(mockCompression.getOutputStream(any(OutputStream.class))).thenReturn(mockOutputStream);
-        when(mockCompression.getInputStream(any(InputStream.class))).thenThrow(new IOException("Compression failed"));
-
-        DocumentPropertyState.setCompressionThreshold(DEFAULT_COMPRESSION_THRESHOLD);
-        DocumentPropertyState documentPropertyState = new DocumentPropertyState(mockDocumentStore, "p", STRING_HUGEVALUE, mockCompression);
-
-        assertEquals(documentPropertyState.getValue(Type.STRING), "{}");
-
-        verify(mockCompression, times(1)).getInputStream(any(InputStream.class));
-    }
-
-    @Test
-    public void defaultValueSetToMinusOne() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException, NoSuchFieldException {
-        DocumentNodeStore store = mock(DocumentNodeStore.class);
-
-        DocumentPropertyState.setCompressionThreshold(-1);
-        DocumentPropertyState state = new DocumentPropertyState(store, "propertyNameNew", "\"" + STRING_HUGEVALUE + "\"", Compression.GZIP);
-
-        byte[] result = state.getCompressedValue();
-
-        assertNull(result);
-        assertEquals(state.getValue(Type.STRING), STRING_HUGEVALUE);
-    }
```

Comment:
This test seems removed. Is there a replacement for it?

Reply:
Done. I added a replacement

Reply:
In CompressedDocumentPropertyState constructor if it throws an exception, it will be caught in factory method and call DocumentPropertyState constructor.